eodag/plugins/authentication/openid_connect.py

```diff
@@ -154,6 +154,8 @@ def authenticate(self) -> Union[AuthBase, Dict[str, str]]:
             exchange_url = user_consent_response.url
         try:
             token = self.exchange_code_for_token(exchange_url, state)
+        except requests.exceptions.Timeout as exc:
+            raise TimeOutError(str(exc))
```

Comment:
please handle exceptions the way it is done in `qssearch`:
```py
raise TimeOutError(exc, timeout=timeout) from exc
```

Reply:
done